libclamav/others_common.c

```diff
@@ -1159,13 +1159,13 @@ char *cli_newfilepath(const char *dir, const char *fname)
     const char *mdir;
     size_t len;
 
+    mdir = dir ? dir : cli_gettmpdir();
+
     if (NULL == fname) {
         cli_dbgmsg("cli_newfilepath('%s'): fname argument must not be NULL\n", mdir);
```

Comment:
Honestly I don't really see the point in printing `mdir` in the debug log here. You could also resolve this by  changing this line to:
```c
cli_dbgmsg("cli_newfilepath: fname argument must not be NULL\n");
```

Related, the only call tho this function is just below on line 1188, and has a bad log message as well, since it's clearly not the only reason this function may fail:
```c
    *name = cli_newfilepath(dir, fname);
    if (!*name) {
        cli_dbgmsg("cli_newfilepathfd('%s'): out of memory\n", dir);
        return CL_EMEM;
    }
```
A better message might be something like:
```c
        cli_dbgmsg("cli_newfilepathfd: failed to create path string for new file '%s' in directory '%s'\n", fname, dir);
```
I also wonder if these error messages shouldn't use `cli_errmsg` instead of `cli_dbgmsg`.